scaaml/capture/input_generators/attack_point_iterator.py

```diff
@@ -71,3 +81,49 @@ def __len__(self) -> int:
 
     def __iter__(self):
         return iter({self._name: value} for value in self._values)
+
+
+class AttackPointIteratorInternalBalancedGenerator(
+        AttackPointIteratorInternalBase):
+    """Attack point terator class that iterates over the balanced generator."""
+
+    def __init__(self, name: str, kwargs: Dict) -> None:
+        """Initialize the balanced kwags to iterate."""
+        self._name = name
+        self._kwargs = kwargs
+        self._len = self._kwargs["length"] * self._kwargs[
+            "bunches"] if "bunches" in self._kwargs else self._kwargs["length"]
+
+    def __len__(self) -> int:
+        return self._len
+
+    def __iter__(self):
+        return iter({self._name: value} for value in balanced_generator(
+            length=self._kwargs["length"],
+            bunches=self._kwargs["bunches"] if "bunches" in self._kwargs else 1,
+            elements=self._kwargs["elements"] if "elements" in
+            self._kwargs else 256))
+
+
+class AttackPointIteratorInternalUnrestrictedGenerator(
+        AttackPointIteratorInternalBase):
+    """
+    Attack point terator class that iterates over the unrestricted generator.
+    """
+
+    def __init__(self, name: str, kwargs: Dict) -> None:
+        """Initialize the unrestricted kwags to iterate."""
+        self._name = name
+        self._kwargs = kwargs
+        self._len = self._kwargs["length"] * self._kwargs[
```

Comment:
Bug here, please add a test for `__len__`.

Reply:
what is the bug?

Reply:
s/length/elements

```
def test_attack_point_iterator_balanced_generator_len():
    config = {
        "operation": "balanced_generator",
        "name": "key",
        "kwargs": {
            "length": 16,
            "bunches": 2,
            "elements": 3
        }
    }
    output = list(iter(AttackPointIterator(config)))
    print(output)
    assert len(
        output) == config["kwargs"]["bunches"] * config["kwargs"]["elements"]
    assert len(output) == len(AttackPointIterator(config))
```